coalib/bearlib/aspects/collections.py

```diff
@@ -30,3 +30,20 @@ def __contains__(self, aspect):
             if issubaspect(aspect, item):
                 return True
         return False
+
+    def get(self, aspect):
+        """
+        Return first item that match or contain an aspect. See
+        :meth:`coalib.bearlib.aspects.aspectbase.get` for further example.
+
+        :param aspect: An aspectclass OR name of an aspect.
+        :return:       An aspectclass OR aspectclass instance, depend on
+                       AspectList content. Return None if no match found.
+        """
+        if isinstance(aspect, str):
+            aspect = coalib.bearlib.aspects[aspect]
+        try:
+            return next(item.get(aspect) for item in self if
```

Comment:
No. `filter` doesn't create a `list` anymore in python 3. It's an iterator itself now, like `itertools.ifilter` in python 2